### PR TITLE
Allow using PyYAML 5.3.x 

### PR DESCRIPTION
**Pull request by wiget**
PyYaml 5.3.1 contains a security fix for CVE-2020-1747.

## PR Checklist

- [x] Wrote a good commit message & description [see guide below].
- [ ] Commit message starts with `[Resolve #issue-number]`.
- [x] Added/Updated unit tests.
- [x] Added/Updated integration tests (if applicable).
- [x] All unit tests (`make test`) are passing.
- [x] Used the same coding conventions as the rest of the project.
- [x] The new code passes flake8 (`make lint`) checks.
- [x] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 install_requirements = [
     "boto3>=1.3,<2.0",
     "click==6.7",
-    "PyYaml==3.13",
+    "PyYaml==5.3.1",
```
Comment (zaro0508) [inline setup.py]: might be better to just match awscli, "PyYAML>=3.10,<5.3"?   https://github.com/aws/aws-cli/blob/ec0660a86df9f13e2f1465d30ea8fd62ceca3e95/setup.py#L35

Comment (wiget) [reply setup.py:16]: Main reason of my change is to prevent using vulnerable PyYAML. Requiring PyYAML older that 5.3.1 does not fulfill this.

Comment (wiget) [reply setup.py]: Anyway, your suggestion applied.

**Comment by wiget** [comment]
Hello @zaro0508 
Please let me know if anything need to be fixed to get this PR merged.

**Comment by zaro0508** [comment]
could you please take a look at this @ngfgrant

**Comment by ngfgrant** [comment]
PyYaml gives me the heebeejeebees. They really mucked up with the versioning.

I did however get PyYaml 5 working in `sceptre-core` (https://github.com/Sceptre/sceptre-core/blob/master/sceptre/file_manager/parsers.py)

My preference would be to replace PyYaml completely with a better maintained package such as rumel.

With regards to this PR - can you explain  the `if/else` in `setup.py` and why there is conditionals in the requirements file. I don't understand. 

Also I maybe wrong but I think `safe_load` will not like Jinja templating. I can recall messing around with this for a while when bumping the `sceptre-core`

**Comment by wiget** [comment]
> With regards to this PR - can you explain the `if/else` in `setup.py` and why there is conditionals in the requirements file. I don't understand.

PyYAML 5.3 dropped support for Python 3.4. If sceptre does not support Python 3.4 then this conditionals can be removed.

> Also I maybe wrong but I think `safe_load` will not like Jinja templating. I can recall messing around with this for a while when bumping the `sceptre-core`

I will test this on Monday and back to you with results.

**Comment by wiget** [comment]
I tested patched Sceptre on my stacks using Jinja templates and everything worked fine.

**Comment by wiget** [comment]
@ngfgrant @zaro0508 What can I do to get this merged?

**Comment by ngfgrant** [comment]
@wiget Sorry I hadn't noticed this was for the v1 branch. We EOL'd v1 last year and I don't have the capacity to maintain both v1 and v2. PyYaml messed up their versioning and this would be a breaking change that would require other changes to be made in v1 for it to work. So unfortunately I am not going to merge this but if you want to fork and patch please feel free to do so.

**Pull request closed without merging**